### PR TITLE
create DDI views

### PR DESCRIPTION
**Pull request by anthonygwalters**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [ ] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207721854334036

### CODE REVIEW

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:1]: is there a real difference between ES/MS & HS that requires a union? iReady would be handled by the left join

File: src/dbt/kipptaf/models/assessments/staging/stg_assessments__academic_goals.sql [hunk 1]
```diff
@@ -1 +1,3 @@
-select *, from {{ source("assessments", "src_assessments__academic_goals") }}
+select g.*, initcap(regexp_extract(s._dbt_source_relation, r'kipp(\w+)_')) as region,
```
Comment (cbini) [inline src/dbt/kipptaf/models/assessments/staging/stg_assessments__academic_goals.sql:1]: spruced up region extract formula

File: src/dbt/kipptaf/models/assessments/staging/stg_assessments__academic_goals.sql [hunk 2]
```diff
@@ -1 +1,3 @@
...
+from {{ source("assessments", "src_assessments__academic_goals") }} as g
+inner join {{ ref("stg_powerschool__schools") }} as s on g.school_id = s.school_number
```
Comment (cbini) [inline src/dbt/kipptaf/models/assessments/staging/stg_assessments__academic_goals.sql:3]: made inner join

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+        '&jasper_prebuilt_id=308',
+        '&fw_form_form_test=1',
+        '&date=logged_in_date',
+        '&matrix_position=2',
+        '&matrix_order=1',
+        '&matrix_group_by=1',
+        '&gradebook_color_coding=1',
+        '&matrix_stretch=1',
+        '#report'
+    ) as illuminate_matrix,
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:159]: ensure all the params are necessary and accurate

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+    on r.home_work_location_powerschool_school_id = w.schoolid
+    and o.observed_at between w.week_start_monday and w.week_end_sunday
+left join
+    {{ ref("stg_people__location_crosswalk") }} as cw
+    on r.home_work_location_name = cw.name
+left join
+    {{ ref("int_people__leadership_crosswalk") }} as lc
+    on r.home_work_location_powerschool_school_id
+    = lc.home_work_location_powerschool_school_id
+where o.observation_type_abbreviation = 'WT'
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:497]: why not make this a separate view?

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql (hunk 3 above)
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:159]: - [ ] ensure all these params are necessary and accurate

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql (hunk 4 above)
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:497]: why shoehorn this into this view? might want to just add additional columns to existing observation details tableau view

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:1]: goals & QBLs are joined on grade & subject for ES/MS and just subject for HS

File: src/dbt/kipptaf/models/assessments/staging/stg_assessments__academic_goals.sql (hunk 1 above)
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/assessments/staging/stg_assessments__academic_goals.sql:1]: 👍🏻

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql
Comment (cbini) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:1]: k might want to refactor the common elements and calcs into a CTE for manageability

Comment (anthonygwalters) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__ddi_dashboard.sql:1]: examples? like the IEP status stuff?

**Pull request merged**